src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/Subscription.cs

```diff
@@ -0,0 +1,36 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using Nethermind.JsonRpc.WebSockets;
+
+namespace Nethermind.JsonRpc.Modules.Subscribe
+{
+    public abstract class Subscription : IDisposable
+    {
+        protected Subscription()
+        {
+            Id = string.Concat("0x", Guid.NewGuid().ToString("N"));
```

Comment:
**Oh I remember now, we were to change the Id mechanism to be more like geth/parity** I think we need to do this as some clients my depend on it being hexadecimal.

Reply:
Uint++.ToHexString()?

Reply:
I discussed with @marcindsobczak and this is ok - same as geth. We want random distribution of subscription id's, not incremental.